src/features/web-compat.js

```diff
@@ -184,6 +192,87 @@ export default class WebCompat extends ContentFeature {
         }
     }
 
+    mediaSessionFix () {
+        try {
+            if (window.navigator.mediaSession) {
+                return
+            }
+
+            this.defineProperty(window.navigator, 'mediaSession', {
+                value: {
+                },
+                writable: true,
+                configurable: true,
+                enumerable: true
+            })
+            this.defineProperty(window.navigator.mediaSession, 'metadata', {
+                value: null,
+                writable: true,
+                configurable: false,
+                enumerable: false
+            })
+            this.defineProperty(window.navigator.mediaSession, 'playbackState', {
+                value: 'none',
+                writable: true,
+                configurable: false,
+                enumerable: false
+            })
+            this.defineProperty(window.navigator.mediaSession, 'setActionHandler', {
+                value: () => {},
+                configurable: true,
+                enumerable: true
+            })
+            this.defineProperty(window.navigator.mediaSession, 'setCameraActive', {
+                value: () => {},
+                configurable: true,
+                enumerable: true
+            })
+            this.defineProperty(window.navigator.mediaSession, 'setMicrophoneActive', {
+                value: () => {},
+                configurable: true,
+                enumerable: true
+            })
+            this.defineProperty(window.navigator.mediaSession, 'setPositionState', {
+                value: () => {},
+                configurable: true,
+                enumerable: true
+            })
+        } catch {
+            // Ignore exceptions that could be caused by conflicting with other extensions
+        }
+    }
+
+    presentationFix () {
+        try {
+            // @ts-expect-error https://app.asana.com/0/1201614831475344/1203979574128023/f
```

Comment:
I'm not sure what the errors are here but we should fix them or explain why they're not relevant.

https://app.asana.com/0/1201614831475344/1203979574128023/f is a placeholder so we could rapidly adopt TS it shouldn't be used on new code.

Reply:
Error for posterity:
```
src/features/web-compat.js:245:34 - error TS2339: Property 'presentation' does not exist on type 'Navigator'.
```

Reply:
> Those were already present throughout web-compat.js

Yeah they're temporary, this message is a TODO when we adopted TS.

> but I get the impression that TS isn't aware of some of these new APIs so it considers them to be undefined objects

Sure it only has what is cross platform and mature.

I would just change the comment to explain that the interface is missing:
```suggestion
            // @ts-expect-error navigator.presentation isn't defined in TS
```